ecl/hql/hqlattr.cpp

```diff
@@ -3776,6 +3776,92 @@ ITypeInfo * setStreamedAttr(ITypeInfo * _type, bool setValue)
     }
 }
 
+//---------------------------------------------------------------------------------
+enum LikelihoodType
+{
+    LikelihoodUnknown,
+    LikelihoodUnlikely,
+    LikelihoodLikely,
+    LikelihoodGivenValue
+};
+
+void getLikelihoodText(StringBuffer & result, IHqlExpression * expr)
+{
+    IHqlExpression * likelihoodExpr = expr->queryProperty(EPlikelihood);
+
+    enum LikelihoodType LikelihoodType = likelihoodExpr ? ((enum LikelihoodType) getIntValue(likelihoodExpr->queryChild(0))) : LikelihoodUnknown;
+    switch(LikelihoodType)
+    {
+    case LikelihoodLikely:
+        result.set("LIKELY");
+        break;
+    case LikelihoodUnlikely:
+        result.set("UNLIKELY");
+        break;
+    case LikelihoodGivenValue:
+    {
+        IValue * value = likelihoodExpr->queryChild(1)->queryValue();
+        if (value)
+        {
+            double probability = value->getRealValue();
+            probability *= 100;
+            result.setf("%3.2f%%", probability);
+            break;
+        } // else drop-through
+    }
+    case LikelihoodUnknown:
+    default:
+        result.set("UNKNOWN");
+    }
+}
+
+static IHqlExpression * evaluatePropLikelihood(IHqlExpression * expr)
+{
+    if (expr->getOperator() != no_filter)
```

Comment:
This isn't really what I was expecting.  This should be a function of an expression - i.e. it should be called on the expression itself, not on the dataset expression.

E.g., 
    queryLikelihood(cond);